examples/wifi-echo/server/esp32/main/ember-print.cpp

```diff
@@ -0,0 +1,64 @@
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "gen/gen_config.h"
+
+#include <support/CHIPLogging.h>
+
+bool emberAfPrintReceivedMessages = true;
+
+using namespace chip::Logging;
+extern "C" {
+
+void emberAfPrint(int category, const char * format, ...)
+{
+    if (format != NULL)
+    {
+        va_list args;
+        va_start(args, format);
+        chip::Logging::LogV(chip::Logging::kLogModule_Zcl, chip::Logging::kLogCategory_Progress, format, args);
+        va_end(args);
+    }
+}
+
+void emberAfPrintln(int category, const char * format, ...)
+{
+    int len = strlen(format);
+    if (format != NULL && len > 0)
```

Comment:
the Log() call already adds a newline

Reply:
Is there a variant to Log that doesn't add this new line? 
If so I should use that in emberAfPrint. I didn't find it. If there isn't one then we should fix that as a separate PR.

Reply:
> Is there a variant to Log that doesn't add this new line?

no, there isn't...

> If so I should use that in emberAfPrint. I didn't find it. If there isn't one then we should fix that as a separate PR.

fix what?

I can think of 3 answers:
1. add a logging API.  this won't work because what ember is expecting is single-threaded access to the print APIs so that multiple calls to print() end up on the same log entry as the next call to println().  I don't think that will work in CHIP's MT environment.
2. change emberPrint() to save the contents in a static buffer until the next call to emberPrintln().  I think that would be part of this PR.
3. remove all calls to emberPrint(), consolidate them into emberPrintln() calls.

Reply:
Fix the fact that Log is adding a new line when it shouldn't be. 

Sounds like #1 isn't feasible. #2 above also seems not a great idea - I worry about the buffer overfilling and losing logs. 
3 could be done, but that is a separate PR.

Reply:
> Fix the fact that Log is adding a new line when it shouldn't be.

Log() is intended to mean "make a log entry", I think.  It need not mean "add a line", but I think all the backends are line oriented.

> 
> Sounds like #1 isn't feasible. #2 above also seems not a great idea - I worry about the buffer overfilling and losing logs.

2 is feasible in the interim before 3.  I don't think we need to worry about overfilling, the calls to print() before println() are on the order of 3 or 4 times, and the content is all really short.

> 3 could be done, but that is a separate PR.

agreed

Reply:
in either case, you can remove the extra work of appending a newline in println(), as it becomes a log entry without that.

Reply:
Happy to remove the extra line